pkg/sql/codegen/optimize/codegen.go

```diff
@@ -15,21 +15,84 @@ package optimize
 
 import (
 	"bytes"
+	"encoding/json"
 	"fmt"
 	"sqlflow.org/sqlflow/pkg/ir"
 	pb "sqlflow.org/sqlflow/pkg/proto"
+	"sqlflow.org/sqlflow/pkg/sql/codegen/attribute"
 	"strings"
 	"text/template"
 )
 
+func checkIsPositiveInteger(i interface{}, name string) error {
+	if i.(int) <= 0 {
```

Comment:
```go
if v, ok := i.(int); !ok || v <= 0 {
```
 
Check https://tour.golang.org/methods/15 for more examples.

Reply:
Done.